src/test/java/io/github/oliviercailloux/y2018/apartments/valuefunction/LinearAVFTest.java

```diff
@@ -0,0 +1,197 @@
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> linearAVF.setWeightRange(Criterion.NB_MIN_NIGHT, Range.closed(-2d, 0d)));
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> linearAVF.setWeightRange(Criterion.TELE, Range.closed(-2d, 0d)));
+  }
+
+  /** Function to test the adaptation to the subjective value weight of a criteria */
+  @Test
+  void adaptWeightTest() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> linearAVF.setWeightRange(Criterion.TELE, Range.closed(-2d, 0d)));
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> linearAVF.setWeightRange(Criterion.TERRACE, Range.closed(0d, 0d)));
+    assertEquals(Range.closed(5d, 15d), linearAVF.getWeightRange(Criterion.TELE));
+    assertEquals(Range.closed(5d, 15d), linearAVF.getWeightRange(Criterion.TERRACE));
+    linearAVF = linearAVF.adaptWeightRange(Criterion.TERRACE, true);
+    LOGGER.info(
+        "************"
+            + linearAVF.getWeightRange(Criterion.TERRACE).lowerEndpoint()
+            + "/"
+            + linearAVF.getWeightRange(Criterion.TERRACE).upperEndpoint());
+    assertEquals(Range.closed(10d, 15d), linearAVF.getWeightRange(Criterion.TERRACE));
+    assertEquals(12.5d, linearAVF.getMiddleOfRange(Criterion.TERRACE));
+    linearAVF = linearAVF.adaptWeightRange(Criterion.TELE, false);
+    assertEquals(Range.closed(5d, 10d), linearAVF.getWeightRange(Criterion.TELE));
+    assertEquals(7.5d, linearAVF.getMiddleOfRange(Criterion.TELE));
+  }
+
+  /** Function to test if the bounds of an interval adapt well when needed */
+  @Test
+  void adaptBoundsTest() {
+    assertThrows(
+        IllegalArgumentException.class, () -> linearAVF.adaptBounds(Criterion.TELE, 0d, true));
+    linearAVF = linearAVF.adaptBounds(Criterion.FLOOR_AREA_TERRACE, 25d, true);
+    LinearValueFunction lvf = linearAVF.getFloorAreaTerraceValueFunction();
+    assertEquals(25d, lvf.getInterval().lowerEndpoint());
+    assertEquals(
+        0.6,
+        linearAVF.getFloorAreaTerraceValueFunction().getSubjectiveValue(a.getFloorAreaTerrace()));
+    linearAVF = linearAVF.adaptBounds(Criterion.NB_BEDROOMS, 8, false);
+    double nbRooms = a.getNbBedrooms();
+    assertEquals(0d, linearAVF.getNbBedroomsValueFunction().getSubjectiveValue(nbRooms));
+    lvf = linearAVF.getNbBedroomsValueFunction();
+    assertEquals(8, lvf.getInterval().upperEndpoint());
+  }
+
+  /** Function to test if a random apartment generated respects some criteria */
+  @Test
+  void getRandomApartmentValueFunctionTest() {
+    ApartmentValueFunction apart = ApartmentValueFunction.getRandomApartmentValueFunction();
+    assertEquals(1d, apart.getFloorAreaValueFunction().getSubjectiveValue(a.getFloorArea()));
+    LinearValueFunction lvf = (LinearValueFunction) apart.getFloorAreaTerraceValueFunction();
+    assertTrue(lvf.getInterval().upperEndpoint() <= 101d);
```

Comment:
Il ne me semble pas pertinent de tester ceci.

Reply:
fixed in 19b2c7362b4b754b7a9a0a10760fb17904863b52